### PR TITLE
Add script to run Nextflow configuration regression tests locally

### PR DESCRIPTION
**Pull request by nwiltsie**
# Description
This PR adds a new `nfconfigtest` script that allows users to run the regression tests locally:

```console
$ ./nfconfigtest --dev --path ~/src/pipeline-recalibrate-BAM/
Looking for tests in pipeline-recalibrate-BAM
#1/2: test/configtest-F16.json
  Failure
  Compare these files and copy any expected changes:
  Original: /Users/nwiltsie/src/pipeline-recalibrate-BAM/test/configtest-F16.json
  Output:   /Users/nwiltsie/src/pipeline-recalibrate-BAM/test/configtest-F16-out.json
#2/2: test/configtest-F32.json
  Success!
1/2 tests passed
```

The script automatically cleans up any of the `-out.json` files that _do_ match the expected results, leaving only the files that a user needs to review.

In addition to the script itself, I changed the `sed` command in the Dockerfile to be a little more robust. The launcher script for Nextflow 24.01.0-edge no longer has double quotes (I'm assuming just to thwart me specifically), so this is just a bit of future-proofing.

```console
$ docker run -it --entrypoint /bin/bash nextflow/nextflow:23.10.0 -c 'grep -o ".\{30\}$" /.nextflow/tmp/launcher/nextflow-one_*/buildkitsandbox/classpath-*' 2>/dev/null
.jar" "nextflow.cli.Launcher"
```

```console
$ docker run -it --entrypoint /bin/bash nextflow/nextflow:24.01.0-edge -c 'grep -o ".\{30\}$" /.nextflow/tmp/launcher/nextflow-one_*/buildkitsandbox/classpath-*' 2>/dev/null
3.0.jar nextflow.cli.Launcher
```

# Checklist

- [x] This PR **does *NOT* contain** Protected Health Information [(PHI)](https://ohrpp.research.ucla.edu/hipaa/). A repo may ***need to be deleted*** if such data is uploaded. <br> Disclosing PHI is a ***major problem***[^1] - Even ***a small leak can be costly***[^2].
  
- [x] This PR **does *NOT* contain** germline genetic data[^3], RNA-Seq, DNA methylation, microbiome or other molecular data[^4].

[^1]: [UCLA Health reaches $7.5m settlement over 2015 breach of 4.5m patient records](https://healthitsecurity.com/news/ucla-health-reaches-7.5m-settlement-over-2015-breach-of-4.5m)
[^2]: [The average healthcare data breach costs $2.2 million, despite the majority of breaches releasing fewer than 500 records.](https://www.ponemon.org/local/upload/file/Sixth%20Annual%20Patient%20Privacy%20%26%20Data%20Security%20Report%20FINAL%206.pdf)
[^3]: [Genetic information is considered PHI.](https://www.genome.gov/about-genomics/policy-issues/Privacy#:~:text=In%202013%2C%20as%20required%20by,genetic%20information%20for%20underwriting%20purposes.)
  [Forensic assays can identify patients with as few as 21 SNPs](https://www.sciencedirect.com/science/article/pii/S1525157817305962)
[^4]: [RNA-Seq](https://www.nature.com/articles/ng.2248), [DNA methylation](https://ieeexplore.ieee.org/document/7958619), [microbiome](https://www.pnas.org/doi/pdf/10.1073/pnas.1423854112), or other molecular data can be used to predict genotypes (PHI) and reveal a patient's identity.

- [x] This PR **does *NOT* contain** other non-plain text files, such as: compressed files, images (*e.g.* `.png`, .`jpeg`), `.pdf`, `.RData`, `.xlsx`, `.doc`, `.ppt`, or other output files.

_&emsp; To automatically exclude such files using a [.gitignore](https://docs.github.com/en/get-started/getting-started-with-git/ignoring-files) file, see [here](https://github.com/uclahs-cds/template-base/blob/main/.gitignore) for example._

- [x] I have read the [code review guidelines](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3187646/Code+Review+Guidelines) and the [code review best practice on GitHub check-list](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List).

- [x] I have set up or verified the `main` branch protection rule following the [github standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3190380/GitHub+Standards#GitHubStandards-Branchprotectionrule) before opening this pull request.

- [x] The name of the branch is meaningful and well formatted following the [standards](https://uclahs-cds.atlassian.net/wiki/spaces/BOUTROSLAB/pages/3189956/Code+Review+Best+Practice+on+GitHub+-+Check+List), using [AD_username (or 5 letters of AD if AD is too long)]-[brief_description_of_branch].

- [x] I have added the major changes included in this pull request to the `CHANGELOG.md` under the next release version or unreleased, and updated the date.

### CODE REVIEW

**Comment by nwiltsie** [comment]
I forgot to add - the script can be downloaded and run independently of this repository (although it won't work properly until f8bf034 is merged, as that changes the exit code).

File: run-nextflow-tests/nfconfigtest [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+
+        # Check the Nextflow version of the test
+        with test.open() as infile:
+            nextflow_version = json.load(infile)["nextflow_version"]
+
+        if nextflow_version != "23.10.0":
+            logging.error(
+                "  ERROR: Only Nextflow version 23.10.0 is supported"
+            )
+            continue
```
Comment (yashpatel6) [inline run-nextflow-tests/nfconfigtest:130]: Would it make sense to eventually update this to check all of the image tags available and then check that the Nextflow version requested is in the tags?

**Review by yashpatel6 — APPROVED** [review]
Looks good! Tested it out and works as expected

File: run-nextflow-tests/nfconfigtest (hunk 1 above)
Comment (nwiltsie) [reply run-nextflow-tests/nfconfigtest:130]: Almost certainly! Right now 23.10.0 is hard-coded all over the place, but I want to handle that better in general.

**Pull request merged**